libsolidity/ast/ASTAnnotations.h

```diff
@@ -45,7 +45,7 @@ namespace solidity
 {
 
 class Type;
-using TypePointer = std::shared_ptr<Type const>;
+using TypePointer = Type const*;
```

Comment:
Actually, do you happen to know, if - given this alias - ``TypePointer x;`` is guaranteed to be zero-initialized :-)? (That is in contrast to plain ``Type const* x;`` which will *not* be zero-initialized) I vaguely remember that there might be a dark corner in the standard about this, but I'm not sure anymore and I can't find it right now...
When I try this experimentally, this seems to be actually true, but I wouldn't trust that without a standard reference, try:
```
    using DefaultZeroIntPtr = int const*;
    int const* a;
    DefaultZeroIntPtr b;
    std::cout << a << std::endl << b << std::endl;
```

I'm mentioning this, because if this is true, then you could skip all member initializations of ``TypePointer`` :-).

Reply:
after short local discussion, we concluded to leave TypePointer var-decls explicitly initialized.